server/test_environment/test_environment.go

```diff
@@ -81,13 +104,18 @@ func GetTestEnv() (*TestEnv, error) {
 		return nil, err
 	}
 	healthChecker := healthcheck.NewHealthChecker("test-server-type")
-	te := &TestEnv{
-		RealEnv: real_environment.NewRealEnv(configurator, healthChecker),
-	}
+	te.RealEnv = real_environment.NewRealEnv(configurator, healthChecker)
 	c, err := memory_cache.NewMemoryCache(1000 * 1000 * 1000 /* 1GB */)
 	if err != nil {
 		return nil, err
 	}
 	te.SetCache(c)
 	return te, nil
 }
+
+func WithConfigOverride(configYAML string) Opt {
+	return func(te *TestEnv) error {
+		te.yamlConfigs = append(te.yamlConfigs, []byte(configYAML))
+		return nil
+	}
+}
```

Comment:
Since the config file also supports flags -- would it work for your usecase to just use https://golang.org/pkg/flag/#Set to override values?

Reply:
IIUC `flag.Set()` would apply the changed flag value to all tests, so I went ahead and added a `flags.Set` utility func under `server/util/testing/flags`, which automatically resets the value back to what it was before the test.